File: sso-glide/src/androidTest/java/it.niedermann.nextcloud.sso.glide/SingleSignOnOriginHeaderTest.java
package it.niedermann.nextcloud.sso.glide;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import com.bumptech.glide.load.model.Headers;

import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.HashMap;

import static it.niedermann.nextcloud.sso.glide.SingleSignOnStreamFetcher.X_HEADER_SSO_ACCOUNT_NAME;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

@RunWith(AndroidJUnit4.class)
public class SingleSignOnOriginHeaderTest {
    @Test
    public void addSpecialHeader() {
        final String testAccountName = "foo@example.com";

        final Headers headers = new SingleSignOnOriginHeader(testAccountName);
        assertEquals(2, headers.getHeaders().size());
        assertTrue(headers.getHeaders().containsKey(X_HEADER_SSO_ACCOUNT_NAME));

        final Headers headersWithExistingHeaders = new SingleSignOnOriginHeader(testAccountName, () -> new HashMap<String, String>() {{
            put("FOO", "BAR");
            put("MEOW", "WOOF");
        }});
        assertEquals(4, headersWithExistingHeaders.getHeaders().size());
        assertTrue(headersWithExistingHeaders.getHeaders().containsKey(X_HEADER_SSO_ACCOUNT_NAME));
    }
}

File: sso-glide/src/androidTest/java/it.niedermann.nextcloud.sso.glide/SingleSignOnOriginHeaderTest.kt
package it.niedermann.nextcloud.sso.glide

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.bumptech.glide.load.model.Headers
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import java.util.*

@RunWith(AndroidJUnit4::class)
class SingleSignOnOriginHeaderTest {
    @Test
    fun addSpecialHeader() {
        val testAccountName = "foo@example.com"
        val headers: Headers = SingleSignOnOriginHeader(testAccountName)
        Assert.assertEquals(2, headers.headers.size.toLong())
        Assert.assertTrue(headers.headers.containsKey(SingleSignOnStreamFetcher.X_HEADER_SSO_ACCOUNT_NAME))
        val headersWithExistingHeaders: Headers = SingleSignOnOriginHeader(testAccountName) {
            object : HashMap<String?, String?>() {
                init {
                    put("FOO", "BAR")
                    put("MEOW", "WOOF")
                }
            }
        }
        Assert.assertEquals(4, headersWithExistingHeaders.headers.size.toLong())
        Assert.assertTrue(headersWithExistingHeaders.headers.containsKey(SingleSignOnStreamFetcher.X_HEADER_SSO_ACCOUNT_NAME))
    }
}